src/url-loader/fs-url-loader.ts

```diff
@@ -58,14 +48,11 @@ export class FSUrlLoader implements UrlLoader {
     });
   }
 
-  getFilePath(url: string): string {
-    const urlObject = parseUrl(url);
-    const pathname =
-        pathlib.normalize(decodeURIComponent(urlObject.pathname || ''));
-    if (!this._isValid(urlObject, pathname)) {
-      throw new Error(`Invalid URL ${url}`);
+  private getFilePath(url: ResolvedUrl): string {
+    if (!this.canLoad(url)) {
+      throw new Error(`FsUrlLoader can not load url: ${url}`);
     }
-    return this.root ? pathlib.join(this.root, pathname) : pathname;
+    return Uri.parse(url).fsPath;
```

Comment:
Can we use the whatwg URL object instead?

Reply:
I'd like to eventually move that direction, but `vscode-uri` has very nice APIs for going from FS Path -> URI and back again, which is also well tested on windows.

I filed https://github.com/Polymer/polymer-analyzer/issues/804 to track this

Reply:
Or were you talking about url objects instead of strings?